### ISSUE TITLE
[v4] Should react-native-reanimated be an optional peer dependency?

### ISSUE THREAD

**Issue by jgornick** [marklawlor/nativewind#889]
If I never use the animated classes, I won't need `react-native-reanimated` to be available and built in my native app. However, I will continuously get unmeet peer dependency warnings until I add it, which means it'll autolinked and built with my app. 

I noticed that it [used to be an optional peer dependency](https://github.com/marklawlor/nativewind/commit/e629fca8ad5e7a6f0118befd91d95678737c387b), but is now required.

How is `react-native-reanimated` different than `react-native-svg` and `react-native-safe-area-context`, which are optional?

Thanks!

**Comment by jgornick** [comment]
Related to `react-native-reanimated`, does it need to be tightly coupled to version `>=3.6.2`? According to the docs, [the APIs should be the same between 2.x and 3.x](https://docs.swmansion.com/react-native-reanimated/docs/guides/migration-from-2.x). 

In order to provide greater adoption, does it make sense to open it up a bit?

**Comment by derek-primumco** [comment]
In my opinion, if someone is bothering to use Nativewind 4, especially in its current pre-release state, then they should definitely be on @latest for React Native Reanimated.

**Comment by jgornick** [comment]
I tend to agree with you, but when working on a production app used by many distributed component teams, it's not as easy as seems to just force latest.

If the API and features of a 3rd party module are the same across many versions ([even majors like react-native-reanimated](https://docs.swmansion.com/react-native-reanimated/docs/guides/migration-from-2.x)), a library consuming that module should support the minimum version via peer deps. This helps greatly with adoption of the library too.

But this thread is more about react-native-reanimated being a peer dependency, regardless of version 😄
